wagtail/admin/wagtail_hooks.py

```diff
@@ -495,7 +494,8 @@ def register_core_features(features):
             {
                 "icon": "h1",
                 "type": "header-one",
-                "description": gettext("Heading %(level)d") % {"level": 1},
+                # FIXME: interpolation causes immediate evaluation
+                "description": _("Heading %(level)d") % {"level": 1},
```

Comment:
I reverted your change to use dedicated strings for each heading for now. I like the idea compared to the alternatives, but it is going to be quite a bit of work to backport the fix to the release branches, as we'll need to update the translation strings too.

I think we should focus on getting the `JavaScriptCatalog` localized in this PR (along with the other quick wins like swapping `gettext` with `gettext_lazy`), and revisit the heading labels issue in a separate PR for the `main` branch.